doc: Updated nRF53 User Guide for NCS 1.3.0

grochu opened the pull request:
- re-phrased the nRF53 User Guide to be more generic,
- reduced the number of direct references to samples,
- added more description for the Bluetooth hci_rpmsg example.

Ref. NCSDK-5119

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/nrfconnect/sdk-nrf?pullRequest=2347) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/nrfconnect/sdk-nrf?pullRequest=2347) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/nrfconnect/sdk-nrf?pullRequest=2347) it.</sub>

grochu commented:
Additional notes:
- I haven't mentioned the nRF RPC since it will not make it to the NCS 1.3.0
- I haven't built the docs locally

kapi-no commented on doc/nrf/ug_nrf5340.rst:
```diff
@@ -81,20 +77,21 @@ The OpenAMP library uses the IPM SHIM layer, which in turn uses the IPC driver i
 *****************
 
-nRF5340 samples consist of two separate images: one that runs on the network core and one that runs on the application core.
+nRF5340 samples consist of two separate images: one that runs on the network core and one that runs on the application core. Most of the samples which support nRF53 are not dedicated exclusively for this device but can also be build for other chips that have the radio peripheral e.g. nRF52 family. The following subsections briefly list these samples and focus more on those, which run exclusively  ...
```
```suggestion
nRF5340 samples usually consist of two separate images: one that runs on the network core and one that runs on the application core. Most of the samples which support nRF53 are not dedicated exclusively for this device but can also be build for other chips that have the radio peripheral e.g. nRF52 family. The following subsections briefly list these samples and focus more on those, which run exclusively on nRF53.
```
I think we support BLE NS images and this case application part consists of two images (spm + app).

grochu replied:
Good point, fixed.

kapi-no commented on doc/nrf/ug_nrf5340.rst:
```diff
@@ -81,20 +77,21 @@ The OpenAMP library uses the IPM SHIM layer, which in turn uses the IPC driver i
 *****************
 
-nRF5340 samples consist of two separate images: one that runs on the network core and one that runs on the application core.
+nRF5340 samples consist of two separate images: one that runs on the network core and one that runs on the application core. Most of the samples which support nRF53 are not dedicated exclusively for this device but can also be build for other chips that have the radio peripheral e.g. nRF52 family. The following subsections briefly list these samples and focus more on those, which run exclusively  ...
```
Please break this long line into a few smaller ones so that we can track changes more easily (I think we use one sentence per line approach)

grochu replied:
Done

kapi-no commented on doc/nrf/ug_nrf5340.rst:
```diff
@@ -113,9 +110,10 @@ Most of these samples should run on the nRF5340 PDK, but not all have been thoro
+Some BLE samples require configuration adjustments to the :ref:`zephyr:bluetooth-hci-rpmsg-sample` sample as described in the `Network samples`_ section.
 
 These samples must be programmed to the application core, in the secure domain.
+To select whether or not to enable the Network core, use :option:`CONFIG_BOARD_ENABLE_CPUNET`.
```
You can add a sentence that this is option is selected by default when radio protocols subsystem / components are used (e.g. BLE). Or would it be too detailed?

grochu replied:
Added

NordicBuilder commented:
**Some checks failed. Please fix and resubmit.**

## Kconfig issues

```
Undefined Kconfig symbols:

warning: undefined symbol FLOAT:

- Referenced at /jenkins_workspace/workspace/latest_fw-nrfconnect-nrf_PR-2347/nrf/subsys/bluetooth/mesh/Kconfig.models:205:

menuconfig BT_MESH_LIGHT_CTRL_SRV_REG
	bool "Lightness Regulator"
	default y
	depends on FLOAT && BT_MESH_LIGHT_CTRL_SRV && BT_MESH && BT
	help
	  Enable the Lightness PI Regulator for controlling the lightness level
	  through a illuminance sensor feedback loop.

- Referenced at /jenkins_workspace/workspace/latest_fw-nrfconnect-nrf_PR-2347/nrf/subsys/spm/Kconfig:51:

config SPM_SERVICE_RNG
	bool "Request random numbers"
	default y
	select NORDIC_SECURITY_BACKEND
	select FLOAT
	depends on !SOC_NRF5340_CPUAPP && SPM_SECURE_SERVICES && IS_SPM
	help
	  The Non-Secure Firmware is not allowed to use the crypto hardware.
	  This service allows it to request random numbers from the SPM.

- Referenced at /jenkins_workspace/workspace/latest_fw-nrfconnect-nrf_PR-2347/nrf/subsys/zigbee/Kconfig:7:

menuconfig ZIGBEE
	bool "Enable Zigbee stack"
	select APP_LINK_WITH_ZBOSS
	imply FLOAT
	imply COUNTER
	imply COUNTER_TIMER3
	imply NRF_802154_RADIO_DRIVER
	imply ENTROPY_GENERATOR
	imply POLL
	imply FLASH
	imply FLASH_PAGE_LAYOUT
	imply FLASH_MAP
	imply MPU_ALLOW_FLASH_WRITE
	depends on SOC_NRF52840 || SOC_NRF52833

- Referenced at /jenkins_workspace/workspace/latest_fw-nrfconnect-nrf_PR-2347/nrf/lib/bsdlib/Kconfig:8:

config BSD_LIBRARY
	bool "Use BSD Socket library for IP/TLS/DTLS"
	select FLOAT
	select FP_SHARING
	imply NET_SOCKETS_OFFLOAD
	depends on TRUSTED_EXECUTION_NONSECURE
	help
	  Use Nordic BSD Socket library.

- Referenced at /jenkins_workspace/workspace/latest_fw-nrfconnect-nrf_PR-2347/nrf/drivers/sensor/sensor_sim/Kconfig:8:

menuconfig SENSOR_SIM
	bool "Sensor simulator"
	select FP_SHARING if FLOAT
	depends on SENSOR
	help
	  Enable sensor simulator.

- Referenced at /jenkins_workspace/workspace/latest_fw-nrfconnect-nrf_PR-2347/nrf/drivers/gps/gps_sim/Kconfig:8:

menuconfig GPS_SIM
	bool "GPS simulator"
	select FP_SHARING if FLOAT
	help
	  Enable GPS simulator.

- Referenced at /jenkins_workspace/workspace/latest_fw-nrfconnect-nrf_PR-2347/nrf/drivers/gps/nrf9160_gps/Kconfig:8:

menuconfig NRF9160_GPS
	bool "nRF9160 GPS driver [experimental]"
	select FP_SHARING if FLOAT
	depends on BSD_LIBRARY
	help
	  Enable nRF9160 GPS driver.


warning: undefined symbol FP_SHARING:

- Referenced at /jenkins_workspace/workspace/latest_fw-nrfconnect-nrf_PR-2347/nrf/lib/bsdlib/Kconfig:8:

config BSD_LIBRARY
	bool "Use BSD Socket library for IP/TLS/DTLS"
	select FLOAT
	select FP_SHARING
	imply NET_SOCKETS_OFFLOAD
	depends on TRUSTED_EXECUTION_NONSECURE
	help
	  Use Nordic BSD Socket library.

- Referenced at /jenkins_workspace/workspace/latest_fw-nrfconnect-nrf_PR-2347/nrf/drivers/sensor/sensor_sim/Kconfig:8:

menuconfig SENSOR_SIM
	bool "Sensor simulator"
	select FP_SHARING if FLOAT
	depends on SENSOR
	help
	  Enable sensor simulator.

- Referenced at /jenkins_workspace/workspace/latest_fw-nrfconnect-nrf_PR-2347/nrf/drivers/gps/gps_sim/Kconfig:8:

menuconfig GPS_SIM
	bool "GPS simulator"
	select FP_SHARING if FLOAT
	help
	  Enable GPS simulator.

- Referenced at /jenkins_workspace/workspace/latest_fw-nrfconnect-nrf_PR-2347/nrf/drivers/gps/nrf9160_gps/Kconfig:8:

menuconfig NRF9160_GPS
	bool "nRF9160 GPS driver [experimental]"
	select FP_SHARING if FLOAT
	depends on BSD_LIBRARY
	help
	  Enable nRF9160 GPS driver.


warning: undefined symbol BT_LL_SW_LEGACY:

- Referenced at /jenkins_workspace/workspace/latest_fw-nrfconnect-nrf_PR-2347/nrf/drivers/entropy/Kconfig:8:

config ENTROPY_CC310
	bool "Arm CC310 RNG driver for Nordic devices"
	default y
	select ENTROPY_HAS_DRIVER
	select ENTROPY_NRF_FORCE_ALT
	depends on (HW_CC310 || (SOC_NRF9160 && SPM)) && ENTROPY_GENERATOR && !BT_LL_SW_LEGACY && !BT_LLL_VENDOR_NORDIC
	help
	  This option enables the Arm CC310 RNG devices in nRF52840 and nRF9160
	  devices. This is dependent on CC310 being enabled in nrf_security.
```



*Tip: The bot edits this comment instead of posting a new one, so you can check the comment's history to see earlier messages.*

grochu commented:
I fixed addressed comments from @kapi-no and removed the statement in line 117 in the "Application samples" section: _"These samples must be programmed to the application core."_ removed _", in the secure domain"_

ru-fu commented:
OK, I'll start over then ...

grochu commented:
> OK, I'll start over then ...

Sorry, should have waited for your commit.

ru-fu commented on doc/nrf/ug_nrf5340.rst:
```diff
@@ -81,41 +81,51 @@ The OpenAMP library uses the IPM SHIM layer, which in turn uses the IPC driver i
 
 Network samples
 ===============
 
```
I changed this section to text instead of a list because it was pointing to one specific sample and a set of samples, but in a list, all items should be similar.
Like this, it is more consistent with the Application samples section as well.

grochu commented:
> I added my commit now. Mostly simplification and making things more consistent.
> Please check and squash if you're okay with the changes.

Looks good, done.

ru-fu commented:
There's a conflict now since https://github.com/nrfconnect/sdk-nrf/pull/1941 went in. Should be okay to just change the reference.

grochu commented:
> There's a conflict now since #1941 went in. Should be okay to just change the reference.

Yes, let's just change the ref. to point to the new example.

ru-fu commented:
Ready to be merged.
